dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -88,17 +76,6 @@
     QueuedChannel(LimitedChannel delegate, int maxQueueSize, DispatcherMetrics metrics) {
         this.delegate = delegate;
         this.queuedCalls = new LinkedBlockingDeque<>(maxQueueSize);
-        this.backgroundScheduler.scheduleWithFixedDelay(
-                () -> {
-                    try {
-                        schedule();
-                    } catch (Exception e) {
-                        log.error("Uncaught exception while scheduling request. This is a programming error.", e);
-                    }
-                },
-                100,
-                100,
-                TimeUnit.MILLISECONDS);
```

Comment:
Deleting this background scheduler means that if there are 10 calls running and 5 more queued and the delegate suddenly decides it has more capacity available but this is _not_ aligned with an existing call starting or finishing then the queue of 5 won't be drained quickly.

A reason we might be able to suddenly gain more capacity is if we live-reloaded a list of urls and a new URL appeared (to which we had zero in flight requests).

Reply:
I’m not sure a 100ms delay is acceptable in that case either, that’s about 300 full requests+responses/thread. I think we need a way for the delegate channel to signal when it’s hungry.

At the moment reloading urls requires us to recreate the entire channel chain, so queued requests cannot be tried against new targets.

Reply:
Oof yep looks like the RefreshingChannelFactory is currently being used as a wrapper around everything:

```java
    private static RefreshingChannelFactory channelFactory(
            Witchcraft witchcraft,
            Refreshable<EmailRuntimeConfiguration> runtime) {
        return new RefreshingChannelFactory(
                runtime.map(c -> c.servicesBlock().orElse(EMPTY_CONF)),
                c -> JavaChannels.create(c, witchcraft.userAgent(), witchcraft.taggedMetrics()));
    }
```

So that actually means that every time something got live-reloaded we'd be re-creating that `dialogue-request-scheduler`!

Probably worth deleting mentions of the 100ms timeout from this class' javadoc then, as it'll be confusing to see a description that's out of sync with the actual impl.